api/base_controller.go

```diff
@@ -55,17 +60,27 @@ type BaseController struct {
 }
 
 // NewController returns a new base controller
-func NewController(repository storage.Repository, resourceBaseURL string, objectType types.ObjectType, objectBlueprint func() types.Object, defaultPageSize, maxPageSize int) *BaseController {
+func NewController(ctx context.Context, repository storage.Repository, resourceBaseURL string, objectType types.ObjectType, objectBlueprint func() types.Object, options *Options, supportsAsync bool) *BaseController {
+	var scheduler *operations.DefaultScheduler
+	if supportsAsync {
+		scheduler = operations.NewScheduler(ctx, repository, options.OperationSettings)
```

Comment:
Perhaps it will be easier to maintain one scheduler per Service Manager.

Reply:
We discussed it and came to the conclusion that there are no significant benefits and that there would actually be more implications of such an approach.